src/plugins/share/public/components/url_panel_content.tsx

```diff
@@ -89,6 +89,7 @@ export class UrlPanelContent extends Component<UrlPanelContentProps, State> {
       url: '',
       anonymousAccessParameters: null,
       showPublicUrlSwitch: false,
+      showWarningButton: true, // this defaults to `true` because `exportUrlAs` defaults to EXPORT_URL_AS_SNAPSHOT
```

Comment:
I figured this made more sense as part of the component state - previously, we were calculating this on every render, which was overkill.